src/OrchardCore.Modules/OrchardCore.Users/Views/RegistrationSettings.Edit.cshtml

```diff
@@ -9,24 +9,6 @@
 <script asp-name="codemirror-addon-mode-multiplex" at="Foot"></script>
 <script asp-name="codemirror-mode-xml" at="Foot"></script>
 
-<div class="mb-3" asp-validation-class-for="UsersCanRegister">
-    <label asp-for="UsersCanRegister" class="form-label">@T["Configure users registration."]</label>
-    <select asp-for="UsersCanRegister" class="form-select">
-        <option value="@UserRegistrationType.NoRegistration" selected="@(Model.UsersCanRegister == UserRegistrationType.NoRegistration)">@T["No Registration"]</option>
-        <option value="@UserRegistrationType.AllowRegistration" selected="@(Model.UsersCanRegister == UserRegistrationType.AllowRegistration)">@T["Allow Registration"]</option>
-        <option value="@UserRegistrationType.AllowOnlyExternalUsers" selected="@(Model.UsersCanRegister == UserRegistrationType.AllowOnlyExternalUsers)">@T["Allow Only External Users"]</option>
-    </select>
-    <span asp-validation-for="UsersCanRegister"></span>
-
-    <div class="hint">
-        <ul>
-            <li>@T["when 'No Registration' is selected, no new users can register"]</li>
-            <li>@T["when 'Allow Registration' is selected, a link is displayed to allow the users to register on the site. If there are external providers, once authenticated the user can register"]</li>
-            <li>@T["when 'Allow Only External Users' is selected, only users authenticated with external providers can register on the site"]</li>
-        </ul>
-    </div>
-</div>
```

Comment:
look at the release notes to see the change. `UserRegistrationType` is no longer available. Instead you can set the new `AllowSiteRegistrations` to true instead. if you need `AllowOnlyExternalUsers` then just enable the new feature instead. so we don't have the types anymore.

Reply:
Yeah, I see the options classes, but just wanted to make sure if e.g. `if`s where switched around or something. Great, then!

Reply:
According to [ln 48 in `ExternalUserMigrations`](https://github.com/OrchardCMS/OrchardCore/pull/16721/files#diff-02b9a309678bcdd3a6949d887f2a512e491f83d871c19b8557b4cab196bd4bb3R48) just enabling the new feature should be enough for something like Entra ID to work, but it isn't. I don't see the equivalent configuration of what was removed above.

I.e.:

- What should I configure _from the UI_ to allow/disallow user registration (while keeping the Users Registration feature enabled, as before)?
- What should I configure _from the UI_ to allow/disallow Entra ID login (after enabling the Entra ID feature)?

I see neither being possible, only from code.
 
Also, previously, only enabling the Users Registration feature wasn't enough to allow users to register, because you had to set this option. While I don't particularly agree with this UX (it being superfluous with the feature enable), now just enabling the Entra ID feature is also enough to register a new local user (i.e. just enabling Entra ID enables local, not just external, user registration too).

Reply:
@Piedone 

> What should I configure from the UI to allow/disallow user registration (while keeping the Users Registration feature enabled, as before)?

There is a new AllowSiteRegistration on the registration settings on the registration UI that you can set.

> What should I configure from the UI to allow/disallow Entra ID login (after enabling the Entra ID feature)?

You should not need anything. Enabling entraID will automatically enable the new feature which will provide everything needed for the external authentication to work "including" the /Register route "without needing to enable the registration feature".

Reply:
I was missing the `AllowSiteRegistration` on the UI so I added it.

Reply:
@Piedone I think for Entra ID to work, you'll need to call [ConfigureMicrosoftAccountSettings](https://github.com/OrchardCMS/OrchardCore/blob/b3c4b4689135498b0afcf687b801b9d149e6f5b3/src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Extensions/OrchardCoreBuilderExtensions.cs#L9-L19) extensions some where in your code. Not great design but this is an existing behavior "not related to this PR". Can you try calling `ConfigureMicrosoftAccountSettings()` on OC builder and see if you'll get the button?

Reply:
It's also mentioned in the docs.

https://docs.orchardcore.net/en/latest/reference/modules/Microsoft.Authentication/#microsoft-account-microsoft-entra-id-azure-active-directory-settings-configuration

Reply:
Have some minor changes to do in the morning

Reply:
fixed.

Reply:
@Piedone `AllowSiteRegistration` was removed. If you need to enable user registration, simply "User Registration" feature. If you enable any identity-provider feature, the External Authentication feature will be enabled by dependency. By default, we enable registration for external users. However, I added a new setting that allows us to disable external registration "if needed".

Reply:
You're right about `ConfigureAzureADSettings` (`ConfigureMicrosoftAccountSettings` is for personal Microsoft accounts), I missed that. With that, Entra ID config and login works.

However, the previous behavior of not allowing registration by default if Users Registration is enabled, only by flipping a switch, should be retained, otherwise this would be a breaking change. We can remove it in 3.0.

Reply:
@Piedone 
This is not a breaking change. The migration already provides settings updates to ensure we don't have any breaking changes.

In the migration, if you already prevent registration, then we disable the User registration feature and enable "DisableExternalRegustration" settings. The External authentications feature will be enabled if any of the providers are enabled. 

If you are running a new tenant, the. DisableExternalRegisteration will be set to false by default.